### PR TITLE
[IMPROVE] Rewrite roomNotFound to React Component

### PR DESCRIPTION
**Pull request by dougfabris**
### before
image
image

### after
image
image

### CODE REVIEW

**Comment by ggazzo** [comment]
check the variation='danger'

File: app/ui-utils/client/lib/openRoom.js [hunk 1]
```diff
@@ -91,9 +91,11 @@ export const openRoom = async function (type, name, render = true) {
 		} catch (error) {
 			c.stop();
 			if (type === 'd') {
-				const result = await callWithErrorHandling('createDirectMessage', ...name.split(', '));
-				if (result) {
+				try {
+					const result = await callWithErrorHandling('createDirectMessage', ...name.split(', '));
```
Comment (tassoevan) [inline app/ui-utils/client/lib/openRoom.js:95]: `callWithErrorHandling` is redundant here as you still have to catch the exception. Better to use `call`:

```suggestion
					const { rid } = await call('createDirectMessage', ...name.split(', '));
```

File: client/views/room/Room/RoomNotFound/RoomNotFound.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+type RoomNotFoundSession = {
+	type: IRoom['t'];
+	name: IRoom['name'];
+	error: Meteor.Error;
+};
+
+const RoomNotFound = (): ReactElement => {
+	const { isMobile } = useLayout();
+	const t = useTranslation();
+	const { name, type } = useSession('roomNotFound') as RoomNotFoundSession;
```
Comment (tassoevan) [inline client/views/room/Room/RoomNotFound/RoomNotFound.tsx:22]: As `error` is ignored at `RoomNotFound`, I wonder if we can pass `name` and `type` as props, using the template's data context. @ggazzo thoughts?

**Review by tassoevan — DISMISSED** [review]

File: client/views/room/Room/RoomNotFound/RoomNotFound.tsx (hunk 2 above)
Comment (dougfabris) [reply client/views/room/Room/RoomNotFound/RoomNotFound.tsx:22]: As we changed the approach here with the design team, we won't use the room data anymore

**Review by gabriellsh — APPROVED** [review]

**Pull request merged**